examples/conv_filter_visualization.py

```diff
@@ -5,135 +5,191 @@
-    x /= (x.std() + K.epsilon())
-    x *= 0.1
-
-    # clip to [0, 1]
-    x += 0.5
-    x = np.clip(x, 0, 1)
-
-    # convert to RGB array
-    x *= 255
-    if K.image_data_format() == 'channels_first':
-        x = x.transpose((1, 2, 0))
-    x = np.clip(x, 0, 255).astype('uint8')
-    return x
-
-
-# build the VGG16 network with ImageNet weights
-model = vgg16.VGG16(weights='imagenet', include_top=False)
-print('Model loaded.')
-
-model.summary()
-
-# this is the placeholder for the input images
-input_img = model.input
-
-# get the symbolic outputs of each "key" layer (we gave them unique names).
-layer_dict = dict([(layer.name, layer) for layer in model.layers[1:]])
-
-
-def normalize(x):
-    # utility function to normalize a tensor by its L2 norm
-    return x / (K.sqrt(K.mean(K.square(x))) + K.epsilon())
-
-
-kept_filters = []
-for filter_index in range(200):
-    # we only scan through the first 200 filters,
-    # but there are actually 512 of them
-    print('Processing filter %d' % filter_index)
-    start_time = time.time()
-
-    # we build a loss function that maximizes the activation
-    # of the nth filter of the layer considered
-    layer_output = layer_dict[layer_name].output
-    if K.image_data_format() == 'channels_first':
-        loss = K.mean(layer_output[:, filter_index, :, :])
-    else:
-        loss = K.mean(layer_output[:, :, :, filter_index])
-
-    # we compute the gradient of the input picture wrt this loss
-    grads = K.gradients(loss, input_img)[0]
-
-    # normalization trick: we normalize the gradient
-    grads = normalize(grads)
-
-    # this function returns the loss and grads given the input picture
-    iterate = K.function([input_img], [loss, grads])
-
-    # step size for gradient ascent
-    step = 1.
-
-    # we start from a gray image with some random noise
-    if K.image_data_format() == 'channels_first':
-        input_img_data = np.random.random((1, 3, img_width, img_height))
-    else:
-        input_img_data = np.random.random((1, img_width, img_height, 3))
-    input_img_data = (input_img_data - 0.5) * 20 + 128
 
-    # we run gradient ascent for 20 steps
-    for i in range(20):
-        loss_value, grads_value = iterate([input_img_data])
-        input_img_data += grads_value * step
+def visualize_layer(model: Model,
+                    layer_name: str,
+                    step: float = 1.,
+                    epochs: int = 20,
+                    output_dim: Tuple[int, int] = (128, 128)) -> None:
+    """ Visualizes the most relevant filters of one conv-layer in a certain model
+        Parameters:
+            model: The model containing layer_name
+            layer_name: The Layer to be visualized: Has to be a part of model
+            step: step size for gradient ascent
+            epochs: Number of iterations for gradient ascent
+            output_dim: [img_width, img_height] The output image dimensions.
+    """
+
+
+    def _normalize(x: Layer) -> Layer:
```

Comment:
These are functions defined inside a function, used once? They should just be regular utilities.